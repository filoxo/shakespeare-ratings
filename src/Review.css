.review {
  max-width: 20rem;
  margin: 1rem auto;
  border-radius: 2px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
  padding: 0.5rem;
}
.reviewContent {
  align-items: center;
  justify-content: center;
  display: flex;
}
.rating {
  background-color: var(--color-main);
  color: #fff;
  font-size: 1.5rem;
  line-height: 2;
  width: 4rem;
  height: 3rem;
  border-radius: 3rem;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.25);
  margin-right: 0.5rem;
  text-align: center;
}
.reviewActions {
  text-align: right;
}
.reviewMoreBtn {
  font-size: 0.8rem;
  background: none;
  border: none;
  padding: 0.3rem;
  color: var(--color-main);
}
.reviewMoreBtn:disabled {
  opacity: 0.8;
}
.reviewBody {
  margin: 1rem 0.5rem 0.5rem;
}
