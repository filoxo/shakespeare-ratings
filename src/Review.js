import React from 'react'
import './Review.css'

export default class Review extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      localeDate: new Date(props.publish_date).toLocaleString(),
    }
  }
  render() {
    const { rating, author, id } = this.props
    const { localeDate } = this.state
    return (
      <div className="review" id={`review-${id}`}>
        <div className="reviewContent">
          <span className="rating">{rating}</span>
          <p>
            Posted by {author} on {localeDate}
          </p>
        </div>
        <div className="reviewActions">
          <button type="button" className="reviewMoreBtn">
            See full comment
          </button>
        </div>
      </div>
    )
  }
}
