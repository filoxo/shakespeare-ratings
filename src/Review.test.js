import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Review from './Review'

describe('<Review/>', () => {
  let props
  beforeEach(() => {
    props = {
      rating: 0.8,
      publish_date: '2016-09-05T23:25:47.642350Z',
      id: '9783221620868',
      author: 'Kaley Schiller',
      loadReviewBody: jest.fn(),
    }
  })
  it('should render', () => {
    const wrapper = shallow(<Review {...props} />)
    expect(wrapper).toBeTruthy()
    const tree = renderer.create(<Review {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should load body on click', () => {
    const wrapper = shallow(<Review {...props} />)
    wrapper.find('.reviewMoreBtn').simulate('click')
    expect(props.loadReviewBody).toHaveBeenCalled()
  })
  it('should display body if exists', () => {
    const body = 'Lorem ipsum dolor amet'
    const wrapper = shallow(<Review {...props} body={body} />)

    expect(wrapper.find('.reviewBody').text()).toEqual(body)
  })
})
