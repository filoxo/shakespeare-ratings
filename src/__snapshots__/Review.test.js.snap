// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review/> should render 1`] = `
<div
  className="review"
  id="review-9783221620868"
>
  <div
    className="reviewContent"
  >
    <span
      className="rating"
    >
      0.8
    </span>
    <p>
      Posted by 
      Kaley Schiller
       on 
      Mon, Sep 5, 2016
    </p>
  </div>
  <div
    className="reviewActions"
  >
    <button
      className="reviewMoreBtn"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      See full comment
    </button>
  </div>
</div>
`;
